### PR TITLE
RunConfigResult unit testing

### PR DESCRIPTION
**Pull request by nv-braf**
Added single and multi-model unit testing to RunConfigResult class

- Single model checking for all methods including boundary condition constraint checking
- Multi-model (2) checking for relevant methods

### CODE REVIEW

File: tests/test_run_config_result.py [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+        self._construct_throughput_with_latency_constraint_rcr()
+
+    def tearDown(self):
+        NotImplemented
+
+    def test_model_name(self):
+        """
+        Test that model_name is correctly returned
+        """
+        self.assertEqual(self.rcr_empty.model_name(),
```
Comment (tgerdesnv) [inline tests/test_run_config_result.py:37]: does this test anything? Those are just empty MagicMocks().  Seems like you would want to explicitly create an RCR with a name passed in to confirm the path from constructor -> 'get' function.  Same for run_config test below

**Review by tgerdesnv — CHANGES_REQUESTED** [review]

File: tests/test_run_config_result.py (hunk 1 above)
Comment (nv-braf) [reply tests/test_run_config_result.py:37]: Yes, it still tests that the value generated by the MagicMock is returned correctly. You don't need to explicitly create a string to ensure the value is checked. For example, if I changed the model_name() return value to "Tim" I would get:

AssertionError: 'Tim' != <MagicMock id='140736174667856'>

Comment (tgerdesnv) [reply tests/test_run_config_result.py:37]: Yeah I get that, and i agree it is testing that model_name() returns _model_name, but that is only half of the equation. I need to know that _model_name is the variable that actually stores the model name information.  I feel like normally you test the set() and get() functions together to prove it is connected all the way through. In this case, the 'set' is the constructor.

Comment (nv-braf) [reply tests/test_run_config_result.py]: Fixed

**Review by tgerdesnv — APPROVED** [review]
